src/test/modelconfig_test.cpp

```diff
@@ -73,43 +73,122 @@ TEST(ModelConfig, getters_setters) {
 
-    ovms::layouts_map_t layouts;
-    layouts["A"] = "layout_A";
-    layouts["B"] = "layout_B";
+    ovms::layouts_map_2_t layouts;
+    layouts["A"] = ovms::LayoutConfiguration{"NCHW", "NHWC"};
+    layouts["B"] = ovms::LayoutConfiguration{"CN", "NC"};
+
+    config.setLayout_2("NHWC");
+    config.setLayouts_2(layouts);
+
+    auto l1 = config.getLayout_2();
+    auto l2 = config.getLayouts_2();
+    EXPECT_EQ(l1.isSet(), false);
+    ASSERT_EQ(l2.count("A"), 1);
+    ASSERT_EQ(l2.count("B"), 1);
+    EXPECT_EQ(l2.find("A")->second.getTensorLayout(), "NCHW");
+    EXPECT_EQ(l2.find("A")->second.getModelLayout(), "NHWC");
+    EXPECT_EQ(l2.find("B")->second.getTensorLayout(), "CN");
+    EXPECT_EQ(l2.find("B")->second.getModelLayout(), "NC");
+
+    config.setLayout_2("NHWC");
+    l1 = config.getLayout_2();
+    l2 = config.getLayouts_2();
+    EXPECT_EQ(l1.isSet(), true);
+    EXPECT_EQ(l1.getTensorLayout(), "NHWC");
+    EXPECT_EQ(l1.getModelLayout(), "NHWC");
+    EXPECT_EQ(l2.size(), 0);
 
-    config.setLayout("NHWC");
-    config.setLayouts(layouts);
+    config.setLayouts_2(layouts);
+    l1 = config.getLayout_2();
+    l2 = config.getLayouts_2();
+    EXPECT_EQ(l1.isSet(), false);
+    EXPECT_EQ(l2.size(), 2);
+}
 
-    auto l1 = config.getLayout();
-    auto l2 = config.getLayouts();
-    EXPECT_EQ(l1, "");
-    EXPECT_THAT(l2, UnorderedElementsAre(
-                        Pair("A", "layout_A"),
-                        Pair("B", "layout_B")));
+TEST(ModelConfig, parseLayoutParam_single) {
+    using namespace ovms;
+    ModelConfig config;
+    // Valid
+    std::string valid_str1 = "";
+    std::string valid_str2 = "nchw";
+    std::string valid_str3 = " Nhwc : ncHW ";
+    std::string valid_str4 = "nC";
+
+    ASSERT_EQ(config.parseLayoutParameter(valid_str1), StatusCode::OK);
+    EXPECT_EQ(config.getLayouts_2().size(), 0);
+    EXPECT_EQ(config.getLayout_2().getTensorLayout(), "");
+    EXPECT_EQ(config.getLayout_2().getModelLayout(), "");
+
+    ASSERT_EQ(config.parseLayoutParameter(valid_str2), StatusCode::OK);
+    EXPECT_EQ(config.getLayouts_2().size(), 0);
+    EXPECT_EQ(config.getLayout_2().getTensorLayout(), "NCHW");
+    EXPECT_EQ(config.getLayout_2().getModelLayout(), "NCHW");
+
+    ASSERT_EQ(config.parseLayoutParameter(valid_str3), StatusCode::OK);
+    EXPECT_EQ(config.getLayouts_2().size(), 0);
+    EXPECT_EQ(config.getLayout_2().getTensorLayout(), "NHWC");
+    EXPECT_EQ(config.getLayout_2().getModelLayout(), "NCHW");
+
+    ASSERT_EQ(config.parseLayoutParameter(valid_str4), StatusCode::OK);
+    EXPECT_EQ(config.getLayouts_2().size(), 0);
+    EXPECT_EQ(config.getLayout_2().getTensorLayout(), "NC");
+    EXPECT_EQ(config.getLayout_2().getModelLayout(), "NC");
 
-    config.setLayout("NHWC");
-    l2 = config.getLayouts();
-    EXPECT_EQ(l2.size(), 0);
+    // Invalid
+    std::vector<std::string> invalid_str{
+        std::string{"nc::nc"},
+        std::string{":nc:nc"},
+        std::string{"nc>nc"},
```

Comment:
Do we have test eg like this:
NC:NCDHW?

Reply:
We do not have such test. Do you mean configuration validation or openvino behavior test?

Reply:
Add unit test

Reply:
Added